### ISSUE TITLE
Microsoft.Insights/scheduledQueryRules always shown as Create

### ISSUE THREAD

**Issue by fulder** [Azure/arm-template-whatif#102]
## Describe the noise

**Resource type** (i.e. Microsoft.Storage/storageAccounts)
Microsoft.Insights/scheduledQueryRules

**apiVersion** (i.e. 2019-04-01)
2018-04-16

**Client (PowerShell, Azure CLI, or API)**
Azure CLI 2.8.0

**Relevant ARM Template code (we only need the resource object for the above `resourceType` and `apiVersion`, but if it's easier you can include the entire template**
```
{
  "name": "myRule",
  "type": "Microsoft.Insights/scheduledQueryRules",
  "apiVersion": "2018-04-16",
  "location": "northeurope",
  "properties": {
    "enabled": "true",
    "source": {
      "query": "['AzureDiagnostics| where ResourceProvider ==\"MICROSOFT.KEYVAULT\"]",
      "dataSourceId": "<WORKSPACE_ID>",
      "queryType": "ResultCount"
    },
    "schedule": {
      "frequencyInMinutes": "15",
      "timeWindowInMinutes": "60"
    },
    "action": {
      "odata.type": "Microsoft.WindowsAzure.Management.Monitoring.Alerts.Models.Microsoft.AppInsights.Nexus.DataContracts.Resources.ScheduledQueryRules.AlertingAction",
      "severity": "1",
      "aznsAction": {
        "actionGroup": "<ACTION_GROUP_ID>"
      },
      "trigger": {
        "thresholdOperator": "GreaterThan",
        "threshold": "1"
      }
    }
  }
}
```

**Expected response (i.e. "I expected no noise since the template has not been modified since the resources were deployed)**
No noise after the resource is deployed and the template not changed

**Current (noisy) response (either include a screenshot of the what-if output, or copy/paste the text)**
```
+ Microsoft.Insights/scheduledQueryRules/myRule [2018-04-16]
       apiVersion:                                  "2018-04-16"
       id:                                          "<GENERATED_ID>"
       location:                                    "northeurope"
       name:                                        "myRule"
       properties.action.aznsAction.actionGroup:    "<ACTION_GROUP_ID>"
       properties.action.odata.type:                "Microsoft.WindowsAzure.Management.Monitoring.Alerts.Models.Microsoft.AppInsights.Nexus.DataContracts.Resources.ScheduledQueryRules.AlertingAction"
       properties.action.severity:                  "1"
       properties.action.trigger.threshold:         "1"
       properties.action.trigger.thresholdOperator: "GreaterThan"
       properties.enabled:                          "true"
       properties.schedule.frequencyInMinutes:      "15"
       properties.schedule.timeWindowInMinutes:     "60"
       properties.source.dataSourceId:              "<WORKSPACE_ID>"
       properties.source.query:                     "AzureDiagnostics| where ResourceProvider =="MICROSOFT.KEYVAULT""
       properties.source.queryType:                 "ResultCount"
       type:                                        "Microsoft.Insights/scheduledQueryRules"
```

**Comment by fulder** [comment]
I'm seeing the same issue (always Create noise) regarding `Microsoft.Insights/metricAlerts` resource

**Comment by tyconsulting** [comment]
I have a similar issue with microsoft.insights/scheduledQueryRules. If I change the query, the What-if API does not pick up the change
![1](https://user-images.githubusercontent.com/11845994/87929896-831ad600-caca-11ea-81fb-322db6292bb0.png)
![2](https://user-images.githubusercontent.com/11845994/87929905-8615c680-caca-11ea-914b-332612092bd6.png)

**Comment by shenglol** [comment]
I could not reproduce the issue. @fulder @tyconsulting Can you help confirm if this is still happening? If yes, do you mind sharing with us the ARM templates you used when invoking What-If?

**Comment by fulder** [comment]
@shenglol you're correct this was hard to reproduce, I tried with more clean ARM templates with only a single `scheduledQueryRules` as well as using my long template with several resources where I saw the bug. I couldn't reproduce it either and after some more debugging I realized I was logged in as a different service principal with only `Microsoft.Insights/scheduledQueryRules/write` permissions while seeing this bug.

This probably made the `what-if` command think this was a totaly new resourse and always mark it for creation. Don't know if this is a known bug? Could be good to get some feedback in console output for the just like the normal deployment CLI command when there are permissions missing, otherwise this will be hard to debug.

The `Microsoft.Insights/metricAlerts` resource bug was related to the same issue. My service principal had  `Microsoft.Insights/metricAlerts/write` permissions, but was missing `Microsoft.Insights/metricAlerts/write`

**Comment by fulder** [comment]
@tyconsulting Could your issue also be related to missing `Microsoft.Insights/scheduledQueryRules/read` permission?

**Comment by alex-frankel** [comment]
@majastrz - we need to consider how we handle not having read permission, but having write permission on a particular resource, and what information we choose to display

**Comment by TimoNordstrom** [comment]
Any progress on this? Having similar issue than @tyconsulting but when using PowerShell with WhatIf. E.g. having updated alert query in microsoft.insights/scheduledQueryRules the output shows "Resource changes: no change."
